src/services/y-websocket.js

```diff
@@ -44,12 +44,22 @@ messageHandlers[messageSync] = (
 ) => {
 	encoding.writeVarUint(encoder, messageSync)
 	const decoderForRemote = decoding.clone(decoder)
+	const pendingStructsBefore = provider.doc.store.pendingStructs
 	const syncMessageType = syncProtocol.readSyncMessage(
 		decoder,
 		encoder,
 		provider.doc,
 		provider,
 	)
+	if (
+		!pendingStructsBefore
+		&& provider.doc.store.pendingStructs
+		&& !encoder.hasContent
+	) {
+		// message could not be processed
+		encoding.writeVarUint(encoder, messageSync)
+		syncProtocol.writeSyncStep1(encoder, provider.doc)
```

Comment:
Looking at the code that handles `resyncInterva` below, there `this.ws.send(encoding.toUint8Array(encoder))` is called after `syncProtocol.writeSynStep1`. I guess it's not needed here because whatever calls this message handler will send pending requests to the websocket provider anyway?